uriget/uriget.go

```diff
@@ -233,3 +239,38 @@ func (o *options) getGit(ctx context.Context, u *url.URL) ([]byte, error) {
 	o.logger.Printf("Read %d bytes from %s", len(buff), filepath.Join(td, subPath))
 	return buff, nil
 }
+
+func (o *options) getOci(ctx context.Context, u *url.URL) ([]byte, error) {
+	parts := strings.Split(u.Host+u.Path, "/")
```

Comment:
I think we can use [ParseReference](https://pkg.go.dev/oras.land/oras-go/v2/registry#ParseReference) to parse the artifact url into a [Reference](https://pkg.go.dev/oras.land/oras-go/v2/registry#Reference).

```go
ref, err := registry.ParseReference(u.Host + u.Path)
if err != nil {
  return nil, fmt.Errorf("can't parse artifact url in a valid reference: %w", err)
}
registry := ref.Registry
repo := ref.Repository
var tag = "latest"
if ref.Reference != "" {
  tag = ref.Reference
}
```

We might get rid of all the parsing stuff from line 244 to line 256.

Reply:
alright